In [1]:
from greedy import Greedy_Solver
from aco import ACO_Solver
from aco2 import ACO_Solver2
from bb import BB_Solver
from LS_SA import LS_LA_Solver
from LS_SA2 import LS_LA_Solver2
from cp import CP_Solver
from ip import IP_Solver
from data import load_data_from_json2,load_data_from_json
import random,math,json
import matplotlib.pyplot as plt
import time  
import numpy as np

In [6]:
def run(Solver,testcases,result,Solvers):
    solver=Solver(testcases['10s-test'])
    result[solver.name]=[]
    for name in testcases:
        testcase=testcases[name]
        if testcase['shelf_num']>50 and Solvers.index(Solver)<3:
            continue
        solver=Solver(testcase)
        start=time.time()
        ans,cost=solver.solve()
        end=time.time()
        runtime=end-start
        result[solver.name].append((cost,runtime))

In [7]:
testcases=load_data_from_json2()
Solvers=[BB_Solver,CP_Solver,IP_Solver,Greedy_Solver,ACO_Solver,LS_LA_Solver,ACO_Solver2]
result={}


In [9]:
run(BB_Solver,testcases,result,Solvers)

Unexpected exception formatting exception. Falling back to standard exception


Traceback (most recent call last):
  File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\IPython\core\interactiveshell.py", line 3526, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "C:\Users\HP\AppData\Local\Temp\ipykernel_27448\743795327.py", line 1, in <module>
    run(BB_Solver,testcases,result,Solvers)
  File "C:\Users\HP\AppData\Local\Temp\ipykernel_27448\3281485135.py", line 10, in run
    ans,cost=solver.solve()
             ^^^^^^^^^^^^^^
  File "d:\python\fund_opt\mini_project\bb.py", line 38, in solve
    self.BB(curr_bound, 1, curr_weight, curr_path, visited, kpi)
  File "d:\python\fund_opt\mini_project\bb.py", line 80, in BB
    self.BB(curr_bound, level + 1, curr_weight, curr_path, visited, kpi)
  File "d:\python\fund_opt\mini_project\bb.py", line 80, in BB
    self.BB(curr_bound, level + 1, curr_weight, curr_path, visited, kpi)
  File "d:\python\fund_opt\mi

In [5]:
run(CP_Solver,testcases,result,Solvers)
run(IP_Solver,testcases,result,Solvers)

In [9]:
run(ACO_Solver,testcases,result,Solvers)

In [4]:
run(LS_LA_Solver,testcases,result,Solvers)

In [5]:
for name in result:
    print(result[name])
    print('\n')

[(148, 0.03441047668457031), (121, 65.08870482444763)]




BB=[(148, 0.03441047668457031), (121, 65.08870482444763)]
CP=[(148.0, 0.15374088287353516), (121.0, 2.8695554733276367)]
IP=[(148.0, 0.12858200073242188), (121.0, 2.309103488922119)]
Greedy=[(217, 0.0), (172, 0.0), (292, 0.0), (783, 0.001230001449584961), (1051, 0.0010085105895996094), (2525, 0.0069997310638427734)]
ACO_Solver=[(148, 6.819920063018799), (128, 15.474803924560547), (220, 32.51289892196655), (752, 83.69358921051025), (1141, 239.22465085983276), (2957, 1362.8865461349487)]
ACO_Solver2=[(148, 6.819920063018799), (130, 15.474803924560547), (285, 32.51289892196655), (808, 83.69358921051025), (1067, 239.22465085983276), (2734, 1362.8865461349487)]
LS_LA_Solver=[(148, 25.24290680885315), (122, 34.32037806510925), (218, 56.21844244003296), (636, 103.94609093666077), (896, 108.00732851028442), (2197, 355.0689618587494)]
